src/main/java/ArraySortedBag.java

```diff
@@ -7,7 +7,7 @@
     private static final int NOT_FOUND = -1;
     protected T[] bag;
     protected int rear;
-    Iterator<T> it = this.iterator();
+    //Iterator<T> it = this.iterator();
```

Comment:
Remove, don't comment out. Comments are for explaining code, not removing dead code. Version control does that substantially better.

There's also several commented out code blocks in `ArrayIndexedBag`. Ice them as well.

Reply:
The trouble is, the commented out code is the stuff being discussed in topic15, and the not commented out stuff is the "improved" implementation from later. 

In other words, the commenting out is more for keeping track of what they already saw.

Reply:
Ok, I figured that was the case for the full methods. At any rate, this line should definitely be removed, not commented out. Iterators are stateful, they don't make much sense as a field here.